app/src/main/java/org/oppia/app/topic/train/TopicTrainFragmentPresenter.kt

```diff
@@ -3,21 +3,99 @@ package org.oppia.app.topic.train
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.Transformations
+import androidx.recyclerview.widget.LinearLayoutManager
 import org.oppia.app.databinding.TopicTrainFragmentBinding
 import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.Topic
+import org.oppia.app.topic.RouteToQuestionPlayerListener
+import org.oppia.app.viewmodel.ViewModelProvider
+import org.oppia.domain.topic.TEST_TOPIC_ID_0
+import org.oppia.domain.topic.TopicController
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.logging.Logger
 import javax.inject.Inject
 
 /** The presenter for [TopicTrainFragment]. */
 @FragmentScope
 class TopicTrainFragmentPresenter @Inject constructor(
-  private val fragment: Fragment
-) {
-  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
+  activity: AppCompatActivity,
+  private val fragment: Fragment,
+  private val logger: Logger,
+  private val topicController: TopicController,
+  private val viewModelProvider: ViewModelProvider<TopicTrainViewModel>
+) : SkillSelector {
+  lateinit var selectedSkillIdList: ArrayList<String>
+
+  private val routeToQuestionPlayerListener = activity as RouteToQuestionPlayerListener
+
+  private lateinit var skillSelectionAdapter: SkillSelectionAdapter
+
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?, skillList : ArrayList<String>): View? {
+    selectedSkillIdList = skillList
     val binding = TopicTrainFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+
+    skillSelectionAdapter = SkillSelectionAdapter(this)
+    binding.skillRecyclerView.apply {
+      layoutManager = LinearLayoutManager(context)
+      adapter = skillSelectionAdapter
+    }
     binding.let {
+      it.viewModel = getTopicTrainViewModel()
       it.lifecycleOwner = fragment
     }
+    getSkillList()
     return binding.root
   }
+
+  private val topicLiveData: LiveData<Topic> by lazy { getTopicList() }
+
+  // TODO(#135): Get this topic-id or get skillList from [TopicFragment].
+  private val topicResultLiveData: LiveData<AsyncResult<Topic>> by lazy {
+    topicController.getTopic(TEST_TOPIC_ID_0)
+  }
+
+  private fun getSkillList() {
```

Comment:
Regarding naming: this isn't actually a traditional getter (it isn't returning anything). Consider instead: subscribeToTopicLiveData()

Reply:
Okay got it. `getSampleMethod` should basically return something. Therefore updating this method with `subscribeToTopicLiveData `